NFIQ2/NFIQ2Algorithm/src/prediction/RandomForestML.cpp

```diff
@@ -123,6 +123,63 @@ NFIQ2::Prediction::RandomForestML::initModule(const std::string &fileName,
 	return hash;
 }
 
+#ifdef __ANDROID__
+std::string
+NFIQ2::Prediction::RandomForestML::initModule(AAssetManager *assets,
+    const std::string &fileName, const std::string &fileHash)
+{
+	if (assets == nullptr) {
+		throw NFIQ2::Exception(NFIQ2::ErrorCode::BadArguments,
+		    "Prediction::RandomForestML::initModule: AAssetManager is null");
+	}
+	if (fileName.empty()) {
+		throw NFIQ2::Exception(NFIQ2::ErrorCode::BadArguments,
+		    "Prediction::RandomForestML::initModule: "
+		    "file name is empty");
+	}
+	if (fileHash.empty()) {
+		throw NFIQ2::Exception(NFIQ2::ErrorCode::BadArguments,
+		    "Prediction::RandomForestML::initModule: "
+		    "hash value is empty");
+	}
+	AAsset *asset = AAssetManager_open(assets, fileName.c_str(),
+	    AASSET_MODE_STREAMING);
+	if (asset == nullptr) {
+		throw NFIQ2::Exception(NFIQ2::ErrorCode::InvalidConfiguration,
+		    "Prediction::RandomForestML::initModule: "
+		    "requested asset not found");
+	}
+	// reading the file with the asset manager
+	std::stringstream ss;
+	char buffer[BUFSIZ + 1]; // BUFSIZ defined in stdio.h
+	int cnt = AAsset_read(asset, buffer, BUFSIZ);
```

Comment:
Similar question to above: if we read the entire buffer ourselves once instead of looping, what are the performance implications? Doesn't the OS buffer the larger read automatically? Something like this seems more C++

```suggestion
	std::vector<char> buf(AAsset_getLength64(asset));
	if (AAsset_read(asset, buf.data(), buf.size()) != buf.size())
	      // error
```

Reply:
Tracing the code, we eventually **have** to get the params into a `std::string` to pass to OpenCV cv::FileStorage. While this is more C++, it duplicates the amount of needed memory. Resolving.